modules/images/webp-uploads/load.php

```diff
@@ -700,6 +700,11 @@ function webp_uploads_update_sources( $metadata, $valid_mime_transforms, $file,
 	$original_directory = pathinfo( $file, PATHINFO_DIRNAME );
 
 	foreach ( $valid_mime_transforms as $targeted_mime ) {
+		// Make sure the path and file exists as those values are being accessed.
+		if ( empty( $main_images[ $targeted_mime ]['path'] ) || empty( $main_images[ $targeted_mime ]['file'] ) ) {
+			continue;
+		}
+
 		// Add sources to original image metadata.
 		$image_file = $main_images[ $targeted_mime ]['path'];
```

Comment:
The current implementation works, but it might be safer to rely on the directory from the saved image path, rather than the original file's directory.

They should be the same, but the one where the saved main image was stored in is more likely to be the one where all the other files are stored in. Basically, instead of assigning `$original_directory` outside of this loop, we could do it here:
```php
$original_directory = pathinfo( $main_images[ $targeted_mime ]['path'], PATHINFO_DIRNAME );
```

Reply:
Agree, just wanted to avoid the need to call `pathinfo` with the assumption that all files including full size and subsizes are stored in the same diretory.